controllers/machinehealthcheck_targets.go

```diff
@@ -197,17 +197,21 @@ func (r *MachineHealthCheckReconciler) getTargetsFromMHC(clusterClient client.Re
 //getMachinesFromMHC fetches Machines matched by the MachineHealthCheck's
 // label selector
 func (r *MachineHealthCheckReconciler) getMachinesFromMHC(mhc *clusterv1.MachineHealthCheck) ([]clusterv1.Machine, error) {
-	selector, err := metav1.LabelSelectorAsSelector(&mhc.Spec.Selector)
+	labelSelector := mhc.Spec.Selector.DeepCopy()
+	labelSelector.MatchLabels[clusterv1.ClusterLabelName] = mhc.Spec.ClusterName
+
+	selector, err := metav1.LabelSelectorAsSelector(labelSelector)
 	if err != nil {
 		return nil, errors.Wrap(err, "failed to build selector")
 	}
 
-	options := client.ListOptions{
-		LabelSelector: selector,
-		Namespace:     mhc.GetNamespace(),
+	options := []client.ListOption{
+		client.MatchingLabelsSelector{Selector: selector},
+		client.InNamespace(mhc.GetNamespace()),
```

Comment:
Might be good to just move these to be arguments to the List call since it removes a layer of indirection.